### DISCUSSION TITLE
Offshore wind turbine park with HVDC Link.

### DISCUSSION THREAD

**Discussion by PowerKylling** [e2nIEE/pandapower#2119] [Q&A]
Hi Panda power community. 
I have just begun working with Panda power (7 days ago), and really enjoys all the features panda power gives the user, regarding Optimal power flow. 

Right now am I trying to make an Offshore wind turbine park with a HVDC Link. Hopefully in time can it look like Tennets 2 GW offshore standard layout. However for now, will I just be happy to get the small 6 bus system, as is shown in the picture attached, to work. 

The main issue with the code, is that when the DC line is implemented inside the system, will the static generator(WT) not produce anything and bus 1 to 4 will therefor not be in operation
![Results of busses ](https://github.com/e2nIEE/pandapower/assets/143074923/f03118ac-a853-4727-bc5e-105cb5e8c029)
![Simple HVDC power plant](https://github.com/e2nIEE/pandapower/assets/143074923/6a08eb65-5b2c-45ba-af61-d2f04f1f3a9f)
. This is evident from the picture attached.  
 
Hope some of you can help me out, and maybe explain what is wrong with the code. 

Best regards 
Jakob 

My code is: 
`

#%% import 
import pandapower as pp 
from pandapower.plotting import simple_plot

#%%  Bygning af netværk

#create empty net
net = pp.create_empty_network()

#%% byg busser 
bus1 = pp.create_bus(net, name="0.69/66kV LV-side Busbar", vn_kv=0.69, type="b")
bus2 = pp.create_bus(net, name="0.69/66kV HV-side Busbar", vn_kv=66, type="b")
bus3 = pp.create_bus(net, name="66/500kV LV-side Busbar", vn_kv=66, type="b")
bus4 = pp.create_bus(net, name="66/500kV HV-side Busbar (PCC)", vn_kv=500, type="b")
bus5 = pp.create_bus(net, name="400/500kV HV-side Busbar", vn_kv=500, type="b")
bus6 = pp.create_bus(net, name="400/500kV lV-side Busbar(POC)", vn_kv=400, type="b")

#%% External Grid 
Eg1=pp.create_ext_grid(net, bus6, vm_pu=1.013,va_degree=0)
#print(net.ext_grid)

#%% Transformer 
Transformertypes = {"PF - 0.69/33kV transformer":             {"sn_mva": 18, "vn_hv_kv": 66, "vn_lv_kv": 0.69, "vk_percent": 6, "vkr_percent": 0.111,"i0_percent":0.24,"pfe_kw": 6, "shift_degree": 150 },
                    "PF - 19/380kV transformer": {"sn_mva": 255, "vn_hv_kv": 500, "vn_lv_kv": 66.0, "vk_percent": 16, "vkr_percent": 1.22,"i0_percent":0.06,"pfe_kw": 60.0, "shift_degree": 0 },}
pp.create_std_types(net, data=Transformertypes, element="trafo")

tr1 = pp.create_transformer(net, bus2, bus1, name ="0.69/66kV transformer", std_type="PF - 0.69/33kV transformer")
tr2 = pp.create_transformer(net, bus4, bus3, name ="66/500kV transformer", std_type="PF - 19/380kV transformer")
tr3 = pp.create_transformer(net, bus5, bus6, name ="400/500kV transformer", std_type="PF - 19/380kV transformer")

#%% Lines
 
linetypes = {"0.69kV": {"r_ohm_per_km": 0.6, "x_ohm_per_km": 0.144, "c_nf_per_km": 144, "max_i_ka": 0.588, "type": "cs"},
             "66kV 800mm^2 Copper Single Point Bonding": {"r_ohm_per_km": 0.0325, "x_ohm_per_km": 0.052, "c_nf_per_km": 305, "max_i_ka": 1.000, "type": "cs"}}
pp.create_std_types(net, data=linetypes, element="line")

#lines
line1 = pp.create_line(net, bus2, bus3, length_km=1, std_type="66kV 800mm^2 Copper Single Point Bonding",  name="66 kV line")

#%%WT
pp.create_sgen(net, bus1, p_mw=200,q_mvar=0, name="static generator")

#%% DC-link
dcl1 = pp.create_dcline(net, name="dc line", from_bus=bus4, to_bus=bus5, p_mw=200, loss_percent=1.0, 
                        loss_mw=0.5, vm_from_pu=1.01, vm_to_pu=1.012, max_p_mw=1000,in_service=True,
                        min_q_from_mvar=0,max_q_from_mvar=100, min_q_to_mvar=0, max_q_to_mvar=100)

#%% Tegn toplologien 

simple_plot(net)

#%%% Beregningerne. 

#%%  Load flow
pp.runpp(net)

print(net.res_bus)
`

**Comment by jkisse** [comment]
Hi @PowerKylling, although it is a late reply: The problem in your setup is that there is no reference voltage in the offshore part. If you model the generator as `gen` with `slack=True` instead of a `sgen`, it should work:
`pp.create_gen(net, bus1, p_mw=200, slack=True)`
